azerothcore/azerothcore-wotlk
# fix(Core/ChatHandler): Prevent messages > 255 characters

I added a check of length 255 of the "msg" variable because the maximum number of characters which a player can send via the WoW client is "255".
The critical issue here consists to send to the server (using some tools and addons) a whisp of more than 255 characters (a lot more) which makes crash the client.

I am not sure if this can break something in the game or in some custom addons (I don't think so).

## CHANGES PROPOSED:
-  add a check to not send a big message to the client


## ISSUES ADDRESSED:
- different servers are affected by this problem and several clients crash everyday

## TESTS PERFORMED:
**before PR**: the client crash

**after PR**: the client does not crash because the packet is blocked


## HOW TO TEST THE CHANGES:
I can't share how and the related tools to make the crash here publicly, but I can share you privately (contact me on Discord).

In the meanwhile, you can review the code and check the normal game, try to send/receive all kind of messages with 2 WoW opened.

### Credits
I have to thanks different reporters and the users who gave me the tools and the related instructions for this that I can't mention actually.




## Target branch(es):




 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
